hub/core/query/test/test_query.py

```diff
@@ -316,3 +314,12 @@ def test_query_bug_transformed_dataset(local_ds):
 
     ds_view = local_ds.filter("classes == 'class_0'", scheduler="threaded")
     np.testing.assert_array_equal(ds_view.classes.numpy()[:, 0], [0] * len(ds_view))
+
+
+def test_view_sample_indices(memory_ds):
```

Comment:
Add a test with filtering as specified by user’s use case 

run two filters and combine the query